lib/transport.dart

```diff
@@ -210,6 +210,11 @@ abstract class TransportStream {
   /// A sink for writing data and/or headers to the remote end.
   StreamSink<StreamMessage> get outgoingMessages;
 
+  /// Set the cancellation handler on this stream.
+  ///
+  /// The handler will be called if the stream receives an RST_STREAM frame.
+  void onCancel(void value(int errorCode));
```

Comment:
If we have support for only one cancel handler, it might be better to make it a setter (i.e. `void set onCancel(void handler(int errorCode))`).

Alternatively we could have a list of registered handlers.

Because otherwise users might think one can install multiple handlers and instead the old one is replaced.